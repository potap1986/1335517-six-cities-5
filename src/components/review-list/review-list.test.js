import React from 'react';
import renderer from 'react-test-renderer';
import ReviewList from './review-list';

jest.mock(`../review-item/review-item`, () => `ReviewItem`);

const reviews = [
  {
    id: 1,
    rating: 1,
    comment: `qwerty`,
    date: `2020-11-05T13:38:44.753Z`,
    localUser: {
      id: 1,
      name: `Mark Goldberg`,
      isPro: true,
      avatarUrl: `url`,
    }
  },
  {
    id: 2,
    rating: 2,
    comment: `qwertmmy`,
    date: `2020-12-05T13:38:44.753Z`,
    localUser: {
      id: 1,
      name: `Mark Goldberg`,
      isPro: true,
      avatarUrl: `url`,
    }
  },
];

describe(`<ReviewList /> render`, () => {

  it(`correctly renders Review List w/ reviews`, () => {
    const tree = renderer
      .create(
          <ReviewList
            reviews={reviews}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
