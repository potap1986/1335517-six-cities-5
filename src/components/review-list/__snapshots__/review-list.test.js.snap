// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewList /> render correctly renders Review List w/ reviews 1`] = `
<ul
  className="reviews__list"
>
  <ReviewItem
    review={
      Object {
        "comment": "qwerty",
        "date": "2020-11-05T13:38:44.753Z",
        "id": 1,
        "localUser": Object {
          "avatarUrl": "url",
          "id": 1,
          "isPro": true,
          "name": "Mark Goldberg",
        },
        "rating": 1,
      }
    }
  />
  <ReviewItem
    review={
      Object {
        "comment": "qwertmmy",
        "date": "2020-12-05T13:38:44.753Z",
        "id": 2,
        "localUser": Object {
          "avatarUrl": "url",
          "id": 1,
          "isPro": true,
          "name": "Mark Goldberg",
        },
        "rating": 2,
      }
    }
  />
</ul>
`;
